azure-communication-ui/azure-communication-ui-demo-app/src/main/java/com/azure/android/communication/ui/callingcompositedemoapp/SettingsActivity.kt

```diff
@@ -0,0 +1,235 @@
+            languageSettingLabelDivider.visibility = View.VISIBLE
+            languageAdapterLayout.visibility = View.VISIBLE
+            isRTLCheckBox.visibility = View.VISIBLE
+            customTranslationCheckBox.visibility = View.VISIBLE
+        }
+        SettingsFeatures.setIsLanguageFeatureEnabled(
+            this,
+            SettingsFeatures.isLanguageFeatureEnabled
+        )
+    }
+
+    private fun updateRTLCheckbox() {
+        val selectedLanguage = getSelectedLanguageValue()
+        val isRTLKey = LANGUAGE_ISRTL_VALUE_SHARED_PREF_KEY + selectedLanguage
+        if (selectedLanguage != null) {
+            isRTLCheckBox.isChecked =
+                sharedPreference.getBoolean(isRTLKey, DEFAULT_RTL_VALUE)
+        }
+    }
+
+    private fun updateCustomStringCheckBox() {
+        customTranslationCheckBox.isChecked = sharedPreference
+            .getBoolean(LANGUAGE_CUSTOM_TRANSLATION_ENABLE, DEFAULT_CUSTOM_TRANSLATION_VALUE)
+    }
+
+    private fun getDeviceLanguage(): String {
+        return Locale.getDefault().displayLanguage
+    }
+
+    private fun setLanguageInSharedPrefForFirstTime() {
+        val deviceLocaleLanguage = getDeviceLanguage()
+        if (isFirstRun()) {
+            for (language in supportedLanguages) {
+                if (language == deviceLocaleLanguage) {
+                    setLanguageValueInSharedPref(language)
+                    break
+                }
+            }
+            if (isFirstRun()) setLanguageValueInSharedPref(DEFAULT_LANGUAGE_VALUE)
+        }
+    }
+
+    private fun setLanguageInAdapter() {
+
+        autoCompleteTextView.setText(
+            sharedPreference.getString(
+                LANGUAGE_ADAPTER_VALUE_SHARED_PREF_KEY,
+                DEFAULT_LANGUAGE_VALUE
+            ),
+            true
+        )
+        languageArrayAdapter.filter.filter(null)
+    }
+
+    private fun isFirstRun(): Boolean {
+        return sharedPreference.getString(LANGUAGE_ADAPTER_VALUE_SHARED_PREF_KEY, LANGUAGE_IS_YET_TOBE_SET).equals(
+            LANGUAGE_IS_YET_TOBE_SET
+        )
+    }
+
+    private fun setLanguageValueInSharedPref(languageValue: String) {
+        sharedPreference.edit().putString(LANGUAGE_ADAPTER_VALUE_SHARED_PREF_KEY, languageValue).apply()
+    }
+
+    private fun getSelectedLanguageValue(): String? {
+        return sharedPreference.getString(
+            LANGUAGE_ADAPTER_VALUE_SHARED_PREF_KEY,
+            DEFAULT_LANGUAGE_VALUE
+        )
+    }
+}
+
+// Shared pref Keys for language & rtl settings
+
+const val LANGUAGE_ADAPTER_VALUE_SHARED_PREF_KEY = "LANGUAGE_ADAPTER_VALUE"
+const val LANGUAGE_ISRTL_VALUE_SHARED_PREF_KEY = "RTL_VALUE_OF_"
+const val LANGUAGE_IS_YET_TOBE_SET = "LANGUAGE_IS_YET_TOBE_SET"
+const val LANGUAGE_CUSTOM_TRANSLATION_ENABLE = "LANGUAGE_CUSTOM_TRANSLATION_ENABLE"
+
+// Shared pref default values for language & rtl settings
+
+const val DEFAULT_LANGUAGE_VALUE = "ENGLISH"
```

Comment:
Do we want to use fully resolved names or language codes (e.g. en, en-US, en-CA)

I think we should be using language codes personally, languages can also have regional variations we may want to support in the future.

Reply:
This name value is default for the dropdown selection and we are using the same default value in sharedpreference. We did not use locale code in dropdown view and for store. So I think for this its alright.